clients/web/src/lib/components/atoms/Drawer/Drawer.svelte

```diff
@@ -0,0 +1,92 @@
+<script lang="ts">
+    import {XCircle} from "@steeze-ui/heroicons";
+    import {Icon} from "@steeze-ui/svelte-icon";
+    import {fly, fade, type FlyParams} from "svelte/transition";
+    // Specific import to prevent circular dependency issues
+    import {Portal} from "$lib/components/abstract";
+    import {setContext} from "svelte";
+    import {DRAWER_CLOSE_CONTEXT, type DrawerPosition} from "./types";
+
+    export let open: boolean = false;
+    export let position: DrawerPosition = "left";
+
+    let flyProps: FlyParams = {};
+    $: switch (position) {
+        case "left":
+        default:
+            flyProps = {x: -100};
+            break;
+        case "right":
+            flyProps = {x: 100};
+            break;
+        case "top":
+            flyProps = {y: -100};
+            break;
+        case "bottom":
+            flyProps = {y: 100};
+            break;
+    }
+
+    function handleKeyPress(e: KeyboardEvent) {
+        if (!open) return;
+        if (e.key !== "Escape") return;
+        open = false;
+    }
+
+    function close() {
+        open = false;
+    }
+
+    setContext(DRAWER_CLOSE_CONTEXT, close);
+</script>
+
+<svelte:body on:keydown={handleKeyPress} />
+{#if open}
+    <Portal>
+        <div
+            class="fixed z-30 h-screen w-screen bg-gray-700/50 backdrop-blur top-0 left-0"
+            transition:fade
+            on:click={close}
+        />
+        <div
+            class="fixed z-40 p-4 overflow-y-auto bg-gray-800 drawer {position} text-gray-200"
+            tabindex="-1"
+            transition:fly={flyProps}
+        >
+            <button
+                type="button"
+                class="text-gray-400 bg-transparent rounded-lg text-sm p-1.5 absolute top-2.5 right-2.5 inline-flex items-center hover:bg-gray-600 hover:text-white"
```

Comment:
Since this is absolutely positioned, I think it may overlap with the content placed in the `slot`. Can you make a story with some lorem ipsum and make sure the spacing looks good all the way around?